modules/shapefile/src/lib/parsers/parse-shapefile.ts

```diff
@@ -64,81 +70,100 @@ export async function* parseShapefileInBatches(
     }
   }
 
-  let iterator: any;
+  // TODO: fix type here
+  let iterator;
   if (propertyIterable) {
     iterator = zipBatchIterators(shapeIterable, propertyIterable);
   } else {
     iterator = shapeIterable;
   }
 
   for await (const item of iterator) {
-    let geometries: any;
-    let properties: any;
+    let geometries: BinaryGeometry[];
+    let properties: ObjectRowTable | undefined;
     if (!propertyIterable) {
       geometries = item;
     } else {
       [geometries, properties] = item;
     }
 
-    const geojsonGeometries = parseGeometries(geometries);
+    const geojsonGeometries = parseBinaryGeometriesToGeoJSON(geometries);
     let features = joinProperties(geojsonGeometries, properties);
     if (reproject) {
       // @ts-ignore
       features = reprojectFeatures(features, prj, _targetCrs);
     }
-    yield {
+
+    const shapefileMetadata = {
       encoding: cpg,
       prj,
       shx,
-      header: shapeHeader,
-      data: features
+      header: shapeHeader
+    };
+    yield {
+      batchType: 'data',
+      shape: 'geojson-row-table',
+      length: features.length,
+      data: features,
+      _metadata: {
```

Comment:
Do we send an initial metadata batch? 

If later we stream batches from a worker we don't want too big a metadata object in each batch?

Reply:
We don't currently send an initial metadata batch.